### PR TITLE
Deprecate useless methods in IUnlistedProperty

### PR DESCRIPTION
**Pull request by CrafterKina**
Does any reason exist?

### CODE REVIEW

File: src/main/java/net/minecraftforge/common/property/ExtendedBlockState.java [hunk 1]
```diff
@@ -133,13 +133,14 @@ protected ExtendedStateImplementation(Block block, ImmutableMap<IProperty<?>, Co
...
         @Override
+        @SuppressWarnings("unchecked")
         public <V>V getValue(IUnlistedProperty<V> property)
         {
             if(!this.unlistedProperties.containsKey(property))
             {
                 throw new IllegalArgumentException("Cannot get unlisted property " + property + " as it does not exist in " + getBlock().getBlockState());
             }
-            return property.getType().cast(this.unlistedProperties.get(property).orElse(null));
+            return (V) this.unlistedProperties.get(property).orElse(null);
```
Comment (diesieben07) [inline src/main/java/net/minecraftforge/common/property/ExtendedBlockState.java:143]: This is not equivalent. Your code will cause a crash in whoever is calling this method, this cast is unchecked.

File: src/main/java/net/minecraftforge/common/property/IUnlistedProperty.java [hunk 2]
```diff
@@ -25,7 +25,15 @@
 
     boolean isValid(V value);
 
-    Class<V> getType();
+    @Deprecated
+    default Class<V> getType()
+    {
+        return (Class<V>) Object.class;
+    }
```
Comment (diesieben07) [inline src/main/java/net/minecraftforge/common/property/IUnlistedProperty.java:32]: This is not helpful at all.

**Review by diesieben07 — CHANGES_REQUESTED** [review]
These methods are by no means useless. Why do you think they are? What you did is not equivalent.

File: src/main/java/net/minecraftforge/common/property/ExtendedBlockState.java (hunk 1 above)
Comment (CrafterKina) [reply src/main/java/net/minecraftforge/common/property/ExtendedBlockState.java:143]: ~~property.getType().cast(...) is also unchecked, without property.getType().isInstance(...).~~
`property.getType().cast(...)` can throw `ClassCastException` (and the "generic cast" can't do it), but `withProperty` says the generic type must be `V`, so `ClassCastException` won't be thrown.

**Pull request closed without merging**